ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/LgrDB.hs

```diff
@@ -154,34 +146,34 @@ type LedgerDB blk = LedgerDB.LedgerDB (ExtLedgerState blk) (RealPoint blk)
   Initialization
 -------------------------------------------------------------------------------}
 
-data LgrDbArgs m blk = forall h. Eq h => LgrDbArgs {
+data LgrDbArgs m blk = LgrDbArgs {
       lgrTopLevelConfig :: TopLevelConfig blk
-    , lgrHasFS          :: HasFS m h
+    , lgrHasFS          :: SomeHasFS m
     , lgrParams         :: LedgerDbParams
     , lgrDiskPolicy     :: DiskPolicy
     , lgrGenesis        :: m (ExtLedgerState blk)
     , lgrTracer         :: Tracer m (TraceEvent (RealPoint blk))
     , lgrTraceLedger    :: Tracer m (LedgerDB blk)
```

Comment:
Not related with the pr, but do we ever use this tracer?

Reply:
Good question. At first I thought that we indeed no longer use it. However, when I looked at the PR that introduced it (#1156), I realised that we're not using `lgrTraceLedger` but we *are* using `cdbTraceLedger`. I suppose Edsko introduced it to be able to write [`toChainDbArgs`](https://github.com/input-output-hk/ouroboros-network/blob/1c8924b856baf809f193798ff686cdb2c0ed25d2/ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/Args.hs#L214). So we need we can't simply remove `lgrTraceLedger` without extra changes.

At some point in the future, I'd like to clean up the argument handling of the ChainDB (and get rid of those nasty `error`s :roll_eyes:). We can get rid of `lgrTraceLedger` then. A prerequisite for this is #2534.